### PR TITLE
fix: address CLP review comments from PR #135

### PR DESCRIPTION
**Pull request by crnbarr93**
## Summary

Addresses unresolved review comment threads from PR #135 (CLP implementation). Fixes 5 and 6 are handled by PR #152 and excluded here.

### Critical fix
- **Escrow existence in CLP ack** — `ack_add_concentrated_liquidity` now uses `.may_load()` + escrow instantiation pattern (was `.load()` which would panic on new tokens without existing escrows)

### High severity
- **Pool key validation** — `execute_add_concentrated_liquidity` now validates `pool_key` against stored `POOL_KEY` (was missing, unlike remove path)

### Medium severity
- **Liquidity pair validation** — Validates `add_liquidity_msg.liquidity` tokens match `state.pair` before extraction
- **Query pool_key validation** — `query_pool` now validates the `pool_key` parameter

### Low severity
- **Remove unused `router` field** — Removed dead `router: Addr` from concentrated `InstantiateMsg` (was always ignored in favor of `info.sender`)

### Excluded (handled by PR #152)
- ~~Admin guard on `UpdatePositionTokenContract`~~ — PR #152 removes this variant entirely
- ~~Position token ID check in remove liquidity~~ — PR #152 implements this with `.load()` (position token is now guaranteed to exist)

### No code change needed
- **Fee tier validation** — Already done at line 97 via `validate_concentrated_fee_and_spacing`
- **Sender ChainUid** — Router validates chain identity upstream
- **IBC fee entrypoints** — Self-resolved by reviewer

## Test plan
- [x] `cargo check --workspace` — clean compile
- [x] `cargo test -p concentrated_vlp --lib --tests` — 46 passed
- [x] `cargo test -p factory --lib` — 11 passed
- [ ] `cargo test -p tests-integration` — full integration suite

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: contracts/liquidity/factory/src/ibc/ack_and_timeout.rs [hunk 1]
```diff
@@ -651,16 +651,47 @@ fn ack_add_concentrated_liquidity(
...
                 if token_info.token_type.is_voucher() {
                     continue;
                 }
                 let escrow_contract =
-                    TOKEN_TO_ESCROW.load(deps.storage, token_info.token.clone())?;
-                let send_msg = token_info
-                    .token_type
-                    .create_escrow_msg(token_info.amount, escrow_contract)?;
-                res = res.add_message(send_msg);
+                    TOKEN_TO_ESCROW.may_load(deps.storage, token_info.token.clone())?;
```
Comment (joemonem) [inline contracts/liquidity/factory/src/ibc/ack_and_timeout.rs:662]: Would it be useful to add a unit test that covers the case where this returns `None`?

**Review by joemonem — APPROVED** [review]

**Review by SlayerAnsh — APPROVED** [review]
